### PR TITLE
docs: add HCL and artifice examples to vagrant postprocessor docs

### PR DESCRIPTION
**Pull request by SwampDragons**
Partially addresses #9282

### CODE REVIEW

File: website/pages/docs/post-processors/vagrant.mdx [hunk 1]
```diff
@@ -135,6 +199,16 @@ Specify overrides within the `override` configuration by provider name:
 }
 ```
 
+</Tab>
+<Tab heading="HCL2">
+
+```hcl
+## This feature is not implemented in HCL.
```
Comment (SwampDragons) [inline website/pages/docs/post-processors/vagrant.mdx:206]: Am I right that "builder specific overrides" aren't really a thing in HCL? https://www.packer.io/docs/templates/provisioners#build-specific-overrides

File: website/pages/docs/post-processors/vagrant.mdx [hunk 2]
```diff
@@ -113,6 +113,67 @@ more details about certain options in following sections.
...
+
+build {
+  sources = [
+    "source.null.example"
+  ]
+
+  post-processor "artifice" {
+    files = ["output-virtualbox-iso/vbox-example-disk001.vmdk",
+             "output-virtualbox-iso/vbox-example.ovf"]
+  }
```
Comment (azr) [inline website/pages/docs/post-processors/vagrant.mdx:165]: These two examples are not entirely similar, is this okay?

Comment (SwampDragons) [reply website/pages/docs/post-processors/vagrant.mdx:165]: I think the json one also does it serially -- they're inside a set of double-brackets, which makes it a sequence definition, which means they'll run serially.

Comment (azr) [reply website/pages/docs/post-processors/vagrant.mdx:165]: Oh right ! I mistakenly thought that the first entry of the array was out of the array, sorry ! 👓

**Pull request merged**